Add FolderListView

pavan-testpress opened the pull request:
### Changes done
- Added:
  - Templates.
  - django-filter.
  - FolderListView.
  - TestCases
  - fontawesome icon for folder.
  - Pagination.
  - Searching feature.
  - Sort by name, -created date, modified date.
  - checked consistency within pagination, sort and search combination.
  - disable user to access folders list view withiout logged in.

### Reason for the changes
- To list all the folders showing the count of bookmarks present within that folder.
- To give searching, sorting and pagination flexibility to user.

### Stats
- Number of Test Cases - 13
### Guidelines
- [x] PEP8(Python)
- [x] Is each logical block separated with a line break?
- [x] Have you self reviewed this PR in context to the previous PR feedbacks?

pavan-testpress committed:
Add FolderListView

- Added:
  - Templates.
  - django-filter.
  - FolderListView.
  - TestCases
  - fontawesome icon for folder.
  - Pagination.
  - Searching feature.
  - Sort by name, -created date, modified date.
  - checked consistency within pagination, sort and search combination.
  - disable user to access folders list view withiout logged in.

pavan-testpress committed:
Push gitignore, pipfiles properly

bharathwaaj commented on onboarding/apps/bookmarks/filters.py:
```diff
@@ -0,0 +1,20 @@
+        if 'sort' in self.request.GET:
+            sort = self.request.GET['sort']
+            if sort not in ['-created', '-modified', 'name']:
+                sort = 'name'
```
```
sort = self.request.GET.get('sort', 'name')
if sort not in ['-created', '-modified', 'name']:
    sort = 'name'
```

bharathwaaj replied:
Use https://django-filter.readthedocs.io/en/master/ref/filters.html#ordering-filter

bharathwaaj commented on onboarding/apps/bookmarks/templates/bookmarks/folder_list.html:
```diff
@@ -0,0 +1,32 @@
+    {% if is_paginated %}
+      <div class="pagination">
+        {% if page_obj.has_previous%}
+          <button class="btn btn-dark" name="page" value="{{ page_obj.previous_page_number }}" form="filter_form" >Previous</button>
```
Use https://simpleisbetterthancomplex.com/snippet/2016/08/22/dealing-with-querystring-parameters.html

bharathwaaj commented on onboarding/apps/bookmarks/templates/bookmarks/base.html:
```diff
@@ -0,0 +1,35 @@
+<body>
+  <nav class="navbar navbar-light bg-light justify-content-evenly">
+  <a class="navbar-brand">Bookmarks</a>
+  <form class="form-inline" method="GET" style="align-items: unset" id="filter_form">
```
Use https://django-filter.readthedocs.io/en/master/guide/usage.html#the-template

bharathwaaj commented on onboarding/apps/bookmarks/views.py:
```diff
@@ -0,0 +1,29 @@
+        return filtered_data.qs
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        data = super(FolderListView, self).get_context_data()
```
Pass the filter only to the template here.

pavan-testpress committed:
Add django_filter
- Used django filter for sorting.
- Used django filter for searching.
- In template django_filter form is used.
- Added Template Tags

pavan-testpress committed:
Fix templatetag name
